pkg/apis/performanceprofile/v2/performanceprofile_validation.go

```diff
@@ -234,6 +447,28 @@ func (r *PerformanceProfile) validateHugePages() field.ErrorList {
 	return allErrs
 }
 
+func (r *PerformanceProfile) getArchitectureForPerfProfile() (string, error) {
```

Comment:
The validation of a performance profile isn't particularly performance sensitive, but we should probably avoid calling the k8s API each time we need to check the architecture. Is there a way to get the architecture once per invocation of the validation and just access the result each time we need to check the architecture?

Reply:
I've updated this in the latest changes, @MarSik also had the same comment